modules/porous_flow/include/userobjects/PorousFlowDictator.h

```diff
@@ -0,0 +1,175 @@
+   * eg if porflow_vars = 'pwater pgas', and the variables in
+   * the simulation are 'energy pwater pgas shape'
+   * then porflow_var_num(2) = 1
+   */
+  unsigned int porflow_var_num(unsigned int moose_var_num) const;
+
+  /**
+   * returns true if moose_var_num is not a porflow var
+   * @param moose_var_num the MOOSE variable number
+   * eg if porflow_vars = 'pwater pgas', and the variables in
+   * the simulation are 'energy pwater pgas shape'
+   * then not_porflow_var(0) = true, not_porflow_var(1) = false
+   */
+  bool not_porflow_var(unsigned int moose_var_num) const;
+
+  /**
+   * a vector of pointers to VariableValues
+   * @param porflow_var_num the porflow variable number
+   * eg if porflow_vars = 'pwater pgas', then
+   * (*porflow_vals(1))[qp] = pgas evaluated at quadpoint qp
+   * Also porflow_vals(i) = &coupledValue
+   */
+  const VariableValue * porflow_vals(unsigned int porflow_var_num) const;
+
+  /**
+   * a vector of pointers to old VariableValues
+   * @param porflow_var_num the porflow variable number
+   * eg if porflow_vars = 'pwater pgas', then
+   * (*porflow_vals_old(1))[qp] = old pgas evaluated at quadpoint qp
+   * Also porflow_vals_old(i) = &coupledValueOld
+   */
+  const VariableValue * porflow_vals_old(unsigned int porflow_var_num) const;
+
+  /**
+   * a vector of pointers to grad(Variable)
+   * @param porflow_var_num the porflow variable number
+   * eg if porflow_vars = 'pwater pgas', then
+   * (*grad_var(1))[qp] = grad(pgas) evaluated at quadpoint qp
+   * Also grad_var(i) = &coupledGradient
+   */
+  const VariableGradient * grad_var(unsigned int porflow_var_num) const;
+
+  /**
+   * The nodal variable values for the given porflow_var_num
+   * To extract the value of porflow variable "pvar", at
+   * node i, use (*PorousFlowDictator.nodal_var(pvar))[i]
+   * @param porflow_var_num the porflow variable number
+   */
+  const VariableValue * nodal_var(unsigned int porflow_var_num) const;
+
+  /**
+   * The old nodal variable values for the given porflow_var_num
+   * @param porflow_var_num the porflow variable number
+   */
+  const VariableValue * nodal_var_old(unsigned int porflow_var_num) const;
+
+  /**
+   * Dummy pressure variable name for use in derivatives using the
+   * DerivativeMaterialInterface
+   */
+  const VariableName pressureVariableNameDummy() const;
+
+  /**
+   * Dummy saturation variable name for use in derivatives using the
+   * DerivativeMaterialInterface
+   */
+  const VariableName saturationVariableNameDummy() const;
+
+  /**
+   * Dummy temperature variable name for use in derivatives using the
+   * DerivativeMaterialInterface
+   */
+  const VariableName temperatureVariableNameDummy() const;
+
+  /**
+   * Dummy mass fraction variable name for use in derivatives using the
+   * DerivativeMaterialInterface
+   */
+  const VariableName massFractionVariableNameDummy() const;
+
+ protected:
+
+  /// number of porousflow variables
+  unsigned int _num_variables;
```

Comment:
This _needs_ to be `const`! If anything ever changes this value after setup your `not_porflow_var` function will not work anymore!